app/src/main/java/eu/kanade/presentation/manga/components/MangaToolbar.kt

```diff
@@ -47,118 +39,106 @@ fun MangaToolbar(
 
     // For action mode
     actionModeCounter: Int,
+    onCancelActionMode: () -> Unit,
     onSelectAll: () -> Unit,
     onInvertSelection: () -> Unit,
 
     modifier: Modifier = Modifier,
-    backgroundAlphaProvider: () -> Float = titleAlphaProvider,
+    backgroundAlphaProvider: () -> Float,
 ) {
-    Column(
+    val isActionMode by remember(actionModeCounter) {
+        derivedStateOf { actionModeCounter > 0 }
+    }
+    AppBar(
         modifier = modifier,
-    ) {
-        val isActionMode = actionModeCounter > 0
-        TopAppBar(
-            title = {
-                Text(
-                    text = if (isActionMode) actionModeCounter.toString() else title,
-                    maxLines = 1,
-                    overflow = TextOverflow.Ellipsis,
-                    color = LocalContentColor.current.copy(alpha = if (isActionMode) 1f else titleAlphaProvider()),
+        title = title,
+        navigateUp = navigateUp,
+        actionModeCounter = actionModeCounter,
+        onCancelActionMode = onCancelActionMode,
+        actionModeActions = {
+            AppBarActions(
+                persistentListOf(
+                    AppBar.Action(
+                        title = stringResource(MR.strings.action_select_all),
+                        icon = Icons.Outlined.SelectAll,
+                        onClick = onSelectAll,
+                    ),
+                    AppBar.Action(
+                        title = stringResource(MR.strings.action_select_inverse),
+                        icon = Icons.Outlined.FlipToBack,
+                        onClick = onInvertSelection,
+                    ),
+                ),
+            )
+        },
+        actions = {
+            var downloadExpanded by remember { mutableStateOf(false) }
+            if (onClickDownload != null) {
+                val onDismissRequest = { downloadExpanded = false }
+                DownloadDropdownMenu(
+                    expanded = downloadExpanded,
+                    onDismissRequest = onDismissRequest,
+                    onDownloadClicked = onClickDownload,
                 )
```

Comment:
Unnecessary?

Reply:
Why? that is the original code to show the chapters' download drop menu.

Reply:
Seems like I confused this with something else